--- a/Makefile	2021-07-04 19:08:50.000000000 +0000
+++ b/Makefile	2021-09-23 14:09:59.292168666 +0000
@@ -2,8 +2,8 @@
 
 # You can set these variables from the command line, and also
 # from the environment for the first two.
-TARGETDIR = /etc/wireguard-initramfs
-INITRAMFS = /etc/initramfs-tools
+TARGETDIR = $(DESTDIR)/etc/wireguard-initramfs
+INITRAMFS = $(DESTDIR)/etc/initramfs-tools
 
 help:
 	@echo "USAGE:"
@@ -21,8 +21,11 @@
 install: remove_legacy
 	@if ! [ "$(shell id -u)" = 0 ]; then echo "You must be root to perform this action."; exit 1; fi
 	@echo "Installing wireguard-initramfs ..."
-	@apt update && apt install initramfs-tools
 	@mkdir -p "$(TARGETDIR)"
+	@mkdir -p "$(INITRAMFS)/hooks"
+	@mkdir -p "$(INITRAMFS)/scripts"
+	@mkdir -p "$(INITRAMFS)/scripts/init-premount"
+	@mkdir -p "$(INITRAMFS)/scripts/init-bottom"
 	@touch "$(TARGETDIR)/private_key"
 	@chmod 0600 "$(TARGETDIR)/private_key"
 	@cp -v config "$(TARGETDIR)/config"
